src/dotnet/commands/dotnet-test/Program.cs

```diff
@@ -42,15 +42,35 @@ public int DoRun(string[] args)
                     RegisterForParentProcessExit(dotnetTestParams.ParentProcessId.Value);
                 }
 
-                var projectContexts = CreateProjectContexts(dotnetTestParams.ProjectPath);
-
-                var projectContext = projectContexts.First();
-
-                var testRunner = projectContext.ProjectFile.TestRunner;
+                var projectContexts = CreateProjectContexts(dotnetTestParams.ProjectPath, dotnetTestParams.Runtime);
+                var ranTests = false;
+                foreach (var projectContext in projectContexts)
+                {
+                    if (dotnetTestParams.Framework != null && dotnetTestParams.Framework != projectContext.TargetFramework)
+                    {
+                        continue;
+                    }
+
+                    ranTests = true;
+                    var testRunner = projectContext.ProjectFile.TestRunner;
+                    var dotnetTestRunner = _dotnetTestRunnerFactory.Create(dotnetTestParams.Port);
+                    var result = dotnetTestRunner.RunTests(projectContext, dotnetTestParams);
```

Comment:
cc @vijayrkn 

We need to make sure that this will still work for VS. I mean, how will VS report tests results for multiple tfms on its test explorer? VS needs to deal with this new semantics. Also, we probably need to change the messages to include the tfm that the results relate to, which will imply changes to the protocol.